File: day1.swift
import Foundation

do {
    let inputString = try String(contentsOfFile: "input_day1.txt", encoding: .utf8)
    let lines = inputString.split(separator: "\n")

    let columns = lines.map { line -> (Int?, Int?) in
        let vals = line.split(separator: "   ")
        return (Int(vals[0]), Int(vals[1]))
    }

    let leftColumn = columns.compactMap({ $0.0 })
    let rightColumn = columns.compactMap({ $0.1 })

    let distance = zip(leftColumn.sorted(), rightColumn.sorted()).reduce(0) {
        result, vals in
        let (left, right) = vals
        return result + abs(left - right)
    }

    print(distance)

    var counts: [Int: Int] = [:]

    for right in rightColumn {
        counts[right, default: 0] += 1
    }

    let similarity = leftColumn.reduce(0) { result, left in
        result + left * counts[left, default: 0]
    }

    print(similarity)

} catch {
    print("\(error)")
}

File: day1.py
from collections import defaultdict


with open("input_day1.txt", "r") as f:
    input = f.readlines()

left_list = sorted([int(line.split("   ")[0]) for line in input])
right_list = sorted([int(line.split("   ")[1]) for line in input])

sum = 0

for left, right in zip(left_list, right_list):
    sum += abs(left - right)

print(sum)

counts = defaultdict(int)

for right in right_list:
    counts[right] += 1

similarity = 0

for left in left_list:
    similarity += left * counts.get(left, 0)

print(similarity)
